### ISSUE TITLE
Type definitions, Deno support.

### ISSUE THREAD

**Issue by tbremer** [preactjs/preact#3075]
- [ ] Check if updating to the latest Preact version resolves the issue

**Describe the bug**

Because of the way the type definitions in Preact are defined, they are unable to be used within Deno. See: https://github.com/denoland/deno/issues/9852

**To Reproduce**

The issue linked above has a reproduction steps, but I have also created a slim Github repo that reproduces this issue: https://github.com/tbremer/preact-deno-types-error

1. Ensure Deno is available on your system
    - Installation: https://deno.land/#installation
1. Clone the repository
1. Optionally pre-cache the modules with: deno cache --config tsconfig.json deps.ts
1. run the mod.tsx file: deno run --config tsconfig.json mod.tsx
1. Error is presented

**Expected behavior**

JSX IntrinsicElements should be consumable in the Deno runtime. 

**Additional Info**

I don't know if I have all the right context to open an immediately mergable pull request, but I would be happy to open an initial take on the problem and collaborate from there until it's resolved.

**Comment by aadamsx** [comment]
@tbremer I was just today researching/considering using Preact for my next Deno project and came across this post.  Are you unable to actually use Preact with Deno?   From your post here: https://github.com/denoland/deno/issues/9852#issue-837355853 it seems you can't even do a simple render in Deno.

**Comment by aadamsx** [comment]
Have you considered Nanojsx? (it supports Deno out of the box) https://nanojsx.github.io

I guess preact has more actual react support, but if you don't need that then.

**Comment by tbremer** [comment]
Preact is my general goto view library, so I just naturally reached for it. I haven't heard of nanojsx, thanks for sharing.

There are ways around it, just none as great as the library provided types. The easiest solution is to create a `jsx.d.ts` file and delcare the namespace accordingly, then import it into your dependencies.

**jsx.d.ts**
```typescript
declare namespace JSX {
  interface IntrinsicElements {
    [elementName: string]: unknown;
  }
}
```

**deps.ts**
```typescript
// @deno-types="./jsx.d.ts"
import * as preact from "https://cdn.skypack.dev/preact?dts";
export { preact, render };
```

You could also use the type definitions and module from the creator of lucacasonato/dext.ts (https://github.com/lucacasonato/dext.ts/tree/main/deps/preact) as they have modified the typings to work within Deno

**Comment by marvinhagemeister** [comment]
@tbremer Ah so this means we need to declare the JSX type globally.

**Comment by tbremer** [comment]
Oh, I see, and if I am understanding correctly the JSX namespace is currently inside of the Preact namespace?

https://github.com/preactjs/preact/blob/master/src/index.d.ts#L7

**Comment by aadamsx** [comment]
@tbremer does this mean just adding a PR that moves line 7 outside the namespace?

**Comment by tbremer** [comment]
Not sure, I am testing that locally right now. @marvinhagemeister would that affect node TS projects or the React compat shimming?

**Comment by tbremer** [comment]
I did a little bit of digging, simply moving the line `export import` statement from line 7 to outside of the preact namespace did not resolve the issue (perhaps because the whole file is declared as a namespace on line 2?).

I've played around with a couple of ideas, none of which appear to be a simple solution.

I think there is some additional risk beacuse the JSX namespace was purposefully moved _into_ the preact namespace in this issue: https://github.com/preactjs/preact/pull/1448.

I'll be away from my computer for the next couple days camping, but will gladly investigate the issue more if it's not resolved when I get back online.
